### PR TITLE
Enabling publishing of standard error for failed test results.

### PR DESCRIPTION
**Pull request by pagangil**
- Adding support for publishing standard error logs for failed test results as attachments. 
- Standard output logs are already being published as attachments for failed test results.

### CODE REVIEW

**Comment by pagangil** [comment]
@puagarwa @nigurr Please review the pull request for adding failure logs for failed test results. Thanks.

**Comment by puagarwa** [comment]
changes looks good to me.

File: src/Agent.Worker/TestResults/TestRunPublisher.cs [hunk 1]
```diff
@@ -316,6 +324,29 @@ private TestAttachmentRequestModel GetConsoleLogAttachmentRequestModel(string co
             return null;
         }
 
+        private TestAttachmentRequestModel GetStandardErrorAttachmentRequestModel(string stdErr)
+        {
+            Trace.Entering();
+            if (!string.IsNullOrWhiteSpace(stdErr))
```
Comment (nigurr) [inline src/Agent.Worker/TestResults/TestRunPublisher.cs]: invert the condition

File: src/Agent.Worker/TestResults/TestRunPublisher.cs [hunk 2]
```diff
@@ -316,6 +324,29 @@ private TestAttachmentRequestModel GetConsoleLogAttachmentRequestModel(string co
...
+            {
+                string stdErrFileName = "Standard Error Output.log";
```
Comment (nigurr) [inline src/Agent.Worker/TestResults/TestRunPublisher.cs]: Avoid spaces in FileNames

File: src/Test/L0/Worker/TestResults/TestRunPublisherTests.cs [hunk 3]
```diff
@@ -352,6 +352,41 @@ public void PublishConsoleLogIsSuccessful()
...
+
+            TestCaseResultData testResultWithStandardError = new TestCaseResultData();
+            testResultWithStandardError.StandardError = "Publish standard error is successfully logged";
+
+            TestCaseResultData testResultWithNoStandardError = new TestCaseResultData();
+            testResultWithNoStandardError.StandardError = "";
+
+            TestCaseResultData testResultDefault = new TestCaseResultData();
+
+            List<TestCaseResultData> testResults = new List<TestCaseResultData>() { testResultWithStandardError, testResultWithNoStandardError, testResultDefault };
```
Comment (nigurr) [inline src/Test/L0/Worker/TestResults/TestRunPublisherTests.cs]: Can you add one more test case where multiple testResultsWithStandardError?

Comment (pagangil) [reply src/Test/L0/Worker/TestResults/TestRunPublisherTests.cs]: Added

**Pull request merged**